src/transforms.rs

```diff
@@ -348,13 +349,13 @@ impl Transformer {
     /// The optional `kafka_message` must be provided to include well known Kafka properties in the value.
     pub(crate) fn transform<M>(
         &self,
-        value: &mut Value,
+        value: &mut DeserializedMessage,
         kafka_message: Option<&M>,
     ) -> Result<(), TransformError>
```

Comment:
Another option, with respect to my earlier comment about changing `DeserializedMessage`, is to have `transform` return `Result<DeserializedMessage, TransformError`. Then it can take `value: DeserializedMessage`. That will allow us to mutate and return what we mutated, but that is a bigger code change.